client/resource_group/controller/controller.go

```diff
@@ -1092,9 +1113,19 @@ func (gc *groupCostController) onResponse(
 			}
 		}
 	}
+
 	gc.mu.Lock()
+	// record the consumption of the request
 	add(gc.mu.consumption, delta)
+	// record the consumption of the request by store
+	// As the penalty is only counted when the request is completed, so here needs to calculate the write cost which is added in `BeforeKVRequest`
+	for _, calc := range gc.calculators {
+		calc.BeforeKVRequest(delta, req)
```

Comment:
I think it will change the original delta, or can we ignore BeforeKV?  why not also do `add(gc.mu.storeCounter[req.StoreID()], delta)` in onRequestWait.

Reply:
the original delta is a temporary value, it's of no use in the following

Reply:
Oh, I missed it